packages/pilot/src/pages/Reprocess/index.js

```diff
@@ -1,234 +1,251 @@
-import React, { Component, Fragment } from 'react'
+import React, { useState, Fragment } from 'react'
 import PropTypes from 'prop-types'
 import copyToClipBoard from 'clipboard-copy'
 import { connect } from 'react-redux'
 import { translate } from 'react-i18next'
 import { withRouter } from 'react-router-dom'
+import moment from 'moment'
 import {
+  __,
+  always,
   compose,
+  cond,
   either,
+  equals,
+  gte,
   isEmpty,
   isNil,
+  path,
+  pipe,
 } from 'ramda'
 
 import ReprocessContainer from '../../containers/Reprocess'
-import { getResponseErrorMessage } from '../../formatters/apiError'
+import { withError } from '../ErrorBoundary'
 
 const mapStateToProps = ({
-  account: { client },
-}) => ({ client })
+  account: {
+    client,
+    company: {
+      capabilities: {
+        allow_reprocess_without_antifraud: allowReprocessWithoutAntifraud,
+      },
+    } = { capabilities: {} },
+  },
+}) => ({
+  allowReprocessWithoutAntifraud,
+  client,
+})
 
 const enhanced = compose(
   translate(),
   connect(mapStateToProps, null),
-  withRouter
+  withRouter,
+  withError
 )
 
-const isEmptyOrNull = either(isEmpty, isNil)
+const isNilOrEmpty = either(isNil, isEmpty)
 
-const reprocessStepStatuses = {
+const stepStatuses = {
   confirmation: {
     confirmation: 'current',
+    identification: 'success',
     result: 'pending',
   },
-  confirmationError: {
-    confirmation: 'error',
+  identification: {
+    confirmation: 'pending',
+    identification: 'current',
     result: 'pending',
   },
   result: {
     confirmation: 'success',
-    result: 'current',
+    identification: 'success',
+    result: 'success',
   },
   resultError: {
     confirmation: 'success',
+    identification: 'success',
     result: 'error',
   },
 }
 
-class Reprocess extends Component {
-  constructor (props) {
-    super(props)
+const getPreviousStep = cond([
+  [equals('confirmation'), always('identification')],
+  [equals('result', always('confirmation'))],
```

Comment:
Tá certo isso ?
não deveria ser assim:
```suggestion
  [equals('result'), always('confirmation')],
```

Reply:
Boa! Na verdade esta condição nem precisa existir, dado que a tela de resultado só tem dois cenários:
- sucesso: usuário fecha o modal ou vai para a transação reprocessada
- falha: usuário clica para tentar novamente e volta para `identification`

Então removi essa verificação